priam/src/main/java/com/netflix/priam/backupv2/BackupVerificationTask.java

```diff
@@ -88,22 +89,32 @@ public void execute() throws Exception {
                                         backupRestoreConfig.getBackupVerificationSLOInHours(),
                                         ChronoUnit.HOURS),
                         DateUtil.getInstant());
-        Optional<BackupVerificationResult> verificationResult =
-                backupVerification.verifyBackup(
+        List<BackupVerificationResult> verificationResults =
+                backupVerification.verifyAllBackups(
                         BackupVersion.SNAPSHOT_META_SERVICE, false, dateRange);
-        if (!verificationResult.isPresent() || !verificationResult.get().valid) {
+        if (verificationResults.isEmpty()
+                || verificationResults
+                        .stream()
+                        .filter(r -> r.valid)
+                        .collect(Collectors.toList())
+                        .isEmpty()) {
             logger.error(
                     "Not able to find any snapshot which is valid in our SLO window: {} hours",
                     backupRestoreConfig.getBackupVerificationSLOInHours());
             backupMetrics.incrementBackupVerificationFailure();
         } else {
             // verification result is available and is valid.
             // send notification that backup is uploaded.
-            logger.info(
-                    "Sending {} message for backup: {}",
-                    AbstractBackupPath.BackupFileType.SNAPSHOT_VERIFIED,
-                    verificationResult.get().snapshotInstant);
-            backupNotificationMgr.notify(verificationResult.get());
+            verificationResults
+                    .stream()
+                    .forEach(
+                            r -> {
```

Comment:
nit: please pick more descriptive variable rather than `r`

Reply:
Sure I will modify it.